sdk/search/azure-search-documents/samples/async_samples/sample_indexers_operations_async.py

```diff
@@ -0,0 +1,124 @@
+# coding: utf-8
+
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+
+"""
+FILE: sample_indexer_operations_async.py
+DESCRIPTION:
+    This sample demonstrates how to get, create, update, or delete a Indexer.
+USAGE:
+    python sample_indexer_operations_async.py
+
+    Set the environment variables with your own values before running the sample:
+    1) AZURE_SEARCH_SERVICE_ENDPOINT - the endpoint of your Azure Cognitive Search service
+    2) AZURE_SEARCH_API_KEY - your search API key
+"""
+
+import asyncio
+import os
+
+service_endpoint = os.getenv("AZURE_SEARCH_SERVICE_ENDPOINT")
+key = os.getenv("AZURE_SEARCH_API_KEY")
+connection_string = os.getenv("AZURE_STORAGE_CONNECTION_STRING")
+
+from azure.core.credentials import AzureKeyCredential
+from azure.search.documents import (
+    DataSource, DataContainer, DataSourceCredentials, Index, Indexer, SimpleField, edm
+)
+from azure.search.documents.aio import SearchServiceClient
+
+service_client = SearchServiceClient(service_endpoint, AzureKeyCredential(key))
+indexers_client = service_client.get_indexers_client()
+
+async def create_indexer():
+    # create a datasource
+    ds_client = service_client.get_datasources_client()
+    credentials = DataSourceCredentials(connection_string=connection_string)
+    container = DataContainer(name='searchcontainer')
+    ds = DataSource(name="async-indexer-datasource", type="azureblob", credentials=credentials, container=container)
+    async with ds_client:
+        data_source = await ds_client.create_datasource(ds)
+
+    # create an index
+    index_name = "hotels"
+    fields = [
+        SimpleField(name="hotelId", type=edm.String, key=True),
+        SimpleField(name="baseRate", type=edm.Double)
+    ]
+    index = Index(name=index_name, fields=fields)
+    ind_client = service_client.get_indexes_client()
+    async with ind_client:
+        index = await ind_client.create_index(index)
+
+    # [START create_indexer_async]
```

Comment:
I don't see a sample snippet for creating a data source, which is required to even use an indexer. We should either include the lines above about creating a data source as part of this one, or at least have snippets called out in the data source-specific sample file below (which I didn't see).

Reply:
@heaths  isn't that up on line 38?

Reply:
Yes, I meant including in a snippet block using your START/END constructs.

Reply:
We already have snippets for datasource in a different file - I don't understand the rationale behind adding snippets for creating datasource in "indexers_client" documentation. If we do it for datasource, should we also include creating an index in the START/END construct?

Reply:
An indexer requires a data source. It's sole purpose is to index documents from a data source into an index, optionally applying AI enrichments.